### PR TITLE
[style] implemented mobile UI  for application ranking page

### PR DESCRIPTION
**Pull request by saitunguturu**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- used useMediaQuery to detect the size of the window to switch between standard desktop view and mobile view
- center gap between the columns changes based on window size
- 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

image

image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- right now the actual value to detect whether a window should be mobile is if the width is **less than 640 px**

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
[](https://www.figma.com/design/fPzdPfT77bx4wXHBkAQbj6/Adopt-an-Inmate?node-id=6325-1034&m=dev
)

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: components/application/matching/MatchingSelectScreen.tsx [hunk 1]
```diff
@@ -15,37 +20,112 @@ export default function MatchingSelectScreen({
...
+            return (
+              <MobileMatchingCard
+                key={m.id}
+                match={m}
+                rank={currentRank}
+                onSelect={handleRankToggle}
+                onReadMore={handleReadMore}
+              />
+            );
+          })}
```
Comment (jinkang-0) [inline components/application/matching/MatchingSelectScreen.tsx]: Best practice: do not put logic inside rendering code, if you need complicated logic, best to make a sub-component
